packages/forms/src/UIForm/fieldsets/Array/Array.component.js

```diff
@@ -61,7 +62,13 @@ export default class ArrayWidget extends React.Component {
 	onAdd(event) {
 		const arrayMergedSchema = this.props.schema;
 		const defaultValue = arrayMergedSchema.schema.items.type === 'object' ? {} : '';
-		const value = [...this.props.value, defaultValue];
+
+		const itemWidget = widgets[this.props.schema.itemWidget];
+		const currentValue =
+			itemWidget && itemWidget.isCloseable
+				? this.props.value.map(item => ({ ...item, isClosed: true }))
+				: this.props.value;
+		const value = currentValue.concat(defaultValue);
```

Comment:
What do you think of
```javascript
let value = [...this.props.value, defaultValue];
if (get(widgets, `${this.props.schema.itemWidget}.isCloseable`, false)) {
  value = value.map(item => ({ ...item, isClosed: true });
}
```

Reply:
hmm I prefer to add some vars with explicit names to be clearer.
Plus, your code close the new inserted item, to have the same behavior, you have to add a test in your `.map()` to skip the last item.

If my code is a little hard to read it's because of the ternary I think. I'll change it to an `if` (@glorieux will be happy). What do you think ?